mesheryctl/pkg/utils/scripts.go

```diff
@@ -1,172 +1,172 @@
+	}
+
+	extract_ca_crt_from_secret() {
+		echo -e -n "\\nExtracting ca.crt from secret..."
+		kubectl get secret --namespace "${NAMESPACE}" "${SECRET_NAME}" -o json | jq \
+		-r '.data["ca.crt"]' | base64 --decode > "${TARGET_FOLDER}/ca.crt"
+		printf "done"
+	}
+
+	get_user_token_from_secret() {
+		echo -e -n "\\nGetting user token from secret..."
+		USER_TOKEN=$(kubectl get secret --namespace "${NAMESPACE}" "${SECRET_NAME}" -o json | jq -r '.data["token"]' | base64 --decode)
+		printf "done"
+	}
+
+	set_kube_config_values() {
+		context=$(kubectl config current-context)
+		echo -e "\\nSetting current context to: $context"
+
+		CLUSTER_NAME=$(kubectl config get-contexts "$context" | awk '{print $3}' | tail -n 1)
+		echo "Cluster name: ${CLUSTER_NAME}"
+
+		ENDPOINT=$(kubectl config view \
+		-o jsonpath="{.clusters[?(@.name == \"${CLUSTER_NAME}\")].cluster.server}")
+		echo "Endpoint: ${ENDPOINT}"
+
+		# Set up the config
+		echo -e "\\nPreparing k8s-${SERVICE_ACCOUNT_NAME}-${NAMESPACE}-conf"
+		echo -n "Setting a cluster entry in kubeconfig..."
+		kubectl config set-cluster "${CLUSTER_NAME}" \
+		--kubeconfig="${KUBECFG_FILE_NAME}" \
+		--server="${ENDPOINT}" \
+		--certificate-authority="${TARGET_FOLDER}/ca.crt" \
+		--embed-certs=true
+
+		echo -n "Setting token credentials entry in kubeconfig..."
+		kubectl config set-credentials \
+		"${SERVICE_ACCOUNT_NAME}-${NAMESPACE}-${CLUSTER_NAME}" \
+		--kubeconfig="${KUBECFG_FILE_NAME}" \
+		--token="${USER_TOKEN}"
+
+		echo -n "Setting a context entry in kubeconfig..."
+		kubectl config set-context \
+		"${SERVICE_ACCOUNT_NAME}-${NAMESPACE}-${CLUSTER_NAME}" \
+		--kubeconfig="${KUBECFG_FILE_NAME}" \
+		--cluster="${CLUSTER_NAME}" \
+		--user="${SERVICE_ACCOUNT_NAME}-${NAMESPACE}-${CLUSTER_NAME}" \
+		--namespace="${NAMESPACE}"
+
+		echo -n "Setting the current-context in the kubeconfig file..."
+		kubectl config use-context "${SERVICE_ACCOUNT_NAME}-${NAMESPACE}-${CLUSTER_NAME}" \
+		--kubeconfig="${KUBECFG_FILE_NAME}"
+	}
+
+	create_target_folder
+	create_service_account
+	get_secret_name_from_service_account
+	extract_ca_crt_from_secret
+	get_user_token_from_secret
+	set_kube_config_values
+
+	echo -e "\\nAll done! Test with:"
+	echo "KUBECONFIG=${KUBECFG_FILE_NAME} kubectl get pods"
+	echo "you should not have any permissions by default - you have just created the authentication part"
+	echo "You will need to create RBAC permissions"
+	`, SAName, namespc)
+
+	generateCFG := exec.Command("sh", "-c", script)
+	generateCFG.Stdout = os.Stdout
+	generateCFG.Stderr = os.Stderr
+
+	return generateCFG.Run()
+}
+
+// GenerateConfigAKS generates kube config file in /tmp/meshery/kubeconfig.yaml for a AKS cluster
+func GenerateConfigAKS(resourceGroup, clusterName string) error {
+	script := fmt.Sprintf(`
+	set -e
+	set -o pipefail
+
+	TARGET_FOLDER="/tmp/meshery"
+	TARGET_FILE="$TARGET_FOLDER/kubeconfig.yaml"
+
+	Resource_Group="%s"
+	Cluster_Name="%s"
+
+	check_for_az() {
```

Comment:
Do we exit if not found? or is this handled elsewhere?

Reply:
`az version` will error with a "command not found", I assume, but no instructions to users to install the prerequisite.

Reply:
We exit if the command is not found.

Reply:
> `az version` will error with a "command not found", I assume, but no instructions to users to install the prerequisite.

I thought of handling this later by providing an option to install `az` in the mesheryctl utility itself

Reply:
Makes sense, however, in effect, this check doesn't do anything different than if the check wasn't included.

Reply:
done the changes, see if it looks alright!